### PR TITLE
docs: mention the experimental nature of the 0.10 branch

### PR DESCRIPTION
**Pull request by ayamir**
part of #1213.

### CODE REVIEW

**Comment by ayamir** [comment]
But the relevant code for `0.10` branch in install scripts was deleted in #1217.

**Comment by Jint-lzxy** [comment]
> But the relevant code for `0.10` branch in install scripts was deleted in #1217.

Oops, my bad for the misunderstanding! What I originally had in mind was maybe we could also add a reminder in the install script for nightly users about the unstable branch (aka, branch `0.10` atm) so they can decide which version to go for. But I'm cool with both approaches loll

**Comment by ayamir** [comment]
IMO adding note for `0.10` branch in readme is enough if relevant code is deleted in install scripts lol.

File: README.md [hunk 1]
```diff
@@ -49,6 +49,8 @@ Branch info:
 |  0.8   |         nvim 0.8         |
 |  0.7   |         nvim 0.7         |
 
+**NOTE: The 0.10 nightly branch is not stable, so don't submit issues for it. This branch is only for users pursuing new features of nvim-nightly.**
```
Comment (Jint-lzxy) [inline README.md:52]:
```suggestion
> [!IMPORTANT]
> The `0.10` branch is intended for nightly neovim builds and is **not** stable. It typically harbors subtle issues scattered throughout. Therefore, refrain from submitting issues if you happen to encounter them. They will be closed directly unless a viable solution is proposed or included.
```
My proposed changes (with added explanations):
> [!IMPORTANT]
> The `0.10` branch is intended for nightly neovim builds and is **not** stable. It typically harbors subtle issues scattered throughout. Therefore, refrain from submitting issues if you happen to encounter them. They will be closed directly unless a viable solution is proposed or included.

Comment (CharlesChiuGit) [reply README.md:52]: that new markup is so cool!

**Comment by CharlesChiuGit** [comment]
i think it would nice to include it in the issue section.

**Review by CharlesChiuGit — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (Jint-lzxy) [reply README.md]: > this new markup looks super cool!

Haha yeah! Actually when I was browsing through some random repos a few months back and stumbled upon this feature, what really blew my mind was that it had actually been rolled out as early as 2022 (starting to really feel that FOMO vibe lmao). More deets: [community#16925](https://github.com/orgs/community/discussions/16925).

**Comment by Jint-lzxy** [comment]
> i think it would nice to include it in the issue section.

I think maybe we could add this hint to other templates too (I mean `custom_config.yml` and `lsp_issue_report.yml`)?

File: README.md (hunk 1 above)
Comment (CharlesChiuGit) [reply README.md]: yeah i did notice this few weeks ago too, at first i thought it was some html trick lol

**Comment by CharlesChiuGit** [comment]
i just edit it in browser, sorry about that lol

**Review by Jint-lzxy — APPROVED** [review]
LGTM!

**Pull request merged**